pype/plugins/global/publish/extract_jpeg.py

```diff
@@ -60,6 +60,19 @@ def process(self, instance):
             full_input_path = os.path.join(stagingdir, input_file)
             self.log.info("input {}".format(full_input_path))
 
+            decompressed_dir = ''
+            decompress = plugins_lib.should_decompress(full_input_path)
+            if decompress:
+                decompressed_dir = plugins_lib.get_decompress_dir()
```

Comment:
Is there some particular reason to have function for returning `decompressed_dir` outside of `decompress` function? Wouldn't it be easier to use it inside `decompress` function and return the `decompressed_dir` from `decompress` function?

Reply:
For me it looks cleaner, function should do one thing and one thing only. This way you have more control, its more testable, no hidden side effects..imho